charts/argo-cd/templates/argocd-applicationset/deployment.yaml

```diff
@@ -24,12 +24,13 @@ spec:
       {{- include "argo-cd.selectorLabels" (dict "context" . "name" .Values.applicationSet.name) | nindent 6 }}
   template:
     metadata:
-      {{- with (mergeOverwrite (deepCopy .Values.global.podAnnotations) .Values.applicationSet.podAnnotations) }}
       annotations:
+        checksum/cmd-params: {{ include (print $.Template.BasePath "/argocd-configs/argocd-cmd-params-cm.yaml") . | sha256sum }}
```

Comment:
I just wondered if this is Argo CD's responsibility to watch its ConfigMap, not Helm Chart's responsibility. 🤔 
*I didn't investigate Argo CD's code base for argocd-cmd-params-cm ConfigMap, but it watches argocd-cm ConfigMap as far as I remember.

Reply:
Oh I see. Thank you for explaining !